kerascv/layers/iou_similarity.py

```diff
@@ -0,0 +1,85 @@
+import tensorflow as tf
+
+
+class IOUSimilarity(tf.keras.layers.Layer):
```

Comment:
Big question here is why a layer -- what workflows will it be used it? We do already have a IoU metric -- would it make sense for IoU to be a metric and a loss instead of a layer?

Reply:
Nice catch.

Reply:
The more I think about it, this is actually pretty different from what 1) giou loss in addons, 2) iou in keras/metrics is doing. 
Both 1) and 2) require y_true and y_pred, which needs to be the same shape, i.e.,:
y_pred: [n_boxes, 4]
y_true: [n_boxes, 4]
And the metrics/loss from it is [n_boxes] before reduction

However what we are having here is:
box1: [n_boxes_1, 4]
box2: [n_boxes_2, 4]
and we're computing the bipartite similarities, so the result would be [n_boxes_1, n_boxes_2]

More specifically, we will need to call this layer with `call(gt_boxes, anchors)`, where n_boxes_1 is usually 3-10 (which is also why I added ragged support because we don't know how many), but n_boxes_2 is always fixed (e.g., 8732 in SSD300). 
This is really used for different things, so I believe it should be a layer.

Reply:
Do we expect the output of this layer to be fed to other layers? Or is it a final output of the model? If the latter, then maybe it should be a metric instead (to be used instead another layer, for instance).

Reply:
IMO, this should be a metric

Reply:
I think that IOU and derivates could be still interesting also as loss:
https://arxiv.org/abs/1908.03851
https://arxiv.org/abs/1911.08287